File: Arrays/KadanesAlgo.py
class Solution:
    ##Complete this function
    #Function to find the sum of contiguous subarray with maximum sum.
    def maxSubArraySum(self,a,size):
        ##Your code here
        max_end_here = float("-inf")
        max_curr_sum = 0
        for i in range(size):
            max_curr_sum += a[i]
            if max_curr_sum<0:
                max_curr_sum = 0
            if max_curr_sum>max_end_here:
                max_end_here = max_curr_sum
        return max_end_here

File: Arrays/KadanesAlgo.cpp
class Solution{
    public:
    // arr: input array
    // n: size of array
    //Function to find the sum of contiguous subarray with maximum sum.
    int maxSubarraySum(int arr[], int n){
        
        // Your code here
        int end = INT_MIN;
        int curr = 0;
        for(int i = 0;i<n;i++){
            curr = curr+arr[i];
            if (curr>end){
                end = curr;
            }
            if (curr<0){
                curr = 0;
            }
            
            
        }
        return end;
        
    }
};
